modules/vactory_decoupled/modules/vactory_two_factor_authentication/src/Controller/VactoryTwoFactorAuthenticationController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Drupal\vactory_two_factor_authentication\Controller;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactory;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Returns responses for vactory_two_factor_authentication routes.
+ */
+class VactoryTwoFactorAuthenticationController extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * Time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * Config.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $config;
+
+  /**
+   * Constructs a new EventFormMailService.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   Time.
+   */
+  public function __construct(TimeInterface $time, ConfigFactory $config) {
+    $this->time = $time;
+    $this->config = $config;
+  }
+
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('datetime.time'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * Builds the response.
+   */
+  public function sendTwoFactorAuthenticationCode(Request $request) {
+    $email = $request->query->get('email', '');
+    if (empty($email)) {
+      return new JsonResponse([
+        'message' => "Parameter mail is required",
+        'status' => 401,
+      ], 401);
+    }
+    // Load user.
+    $user = user_load_by_mail($email);
+    if (!isset($user)) {
+      return new JsonResponse([
+        'message' => "User doesn't exist",
+        'status' => 401,
+      ], 401);
+    }
+
+    $otpService = \Drupal::service('vactory_otp.send_otp');
+    $config = $this->config
+      ->getEditable('vactory_two_factor_authentication.settings');
+    $code_otp = '';
+    $provider = $config->get('type_2fa');
+    switch ($provider) {
+      case 'mail':
+        $mail = $user->getEmail();
+        $code_otp = $otpService->sendOtpByMail('', $mail, '', '');
+        break;
+
+      case 'phone':
+        $user_phone = $user->get('field_telephone')->value;
+        if (!isset($user) || empty($user_phone)) {
+          return new JsonResponse([
+            'message' => "User phone is not set yet",
+            'status' => 401,
+          ], 401);
+        }
+        $user_phone = preg_replace("/^0/", "212", $user_phone);
```

Comment:
Do not force +212 prefix, make it configurable, use the whole field value coming from the user +212644554455 or +33644556677, make sure you force all phone inputs to be using a prefix